### PR TITLE
Make admin functionality

### PR DESCRIPTION
**Pull request by KiranPrabhakar135**
Pull request includes changes for:
1. Enabling and Disabling an user as admin. (#112 ) in admin's client info page

### CODE REVIEW

File: src/main/java/org/onedatashare/server/controller/UserController.java [hunk 1]
```diff
@@ -59,6 +59,15 @@ public Object performAction(@RequestHeader HttpHeaders headers, @RequestBody Use
         return null;
     }
   }
+  @PutMapping
+  public Object putAction(@RequestHeader HttpHeaders headers, @RequestBody UserAction userAction){
+    switch(userAction.action) {
+      case "enableOrDisableAdmin":
```
Comment (LinusCastelino) [inline src/main/java/org/onedatashare/server/controller/UserController.java]: For clarity, can we set this case as 'updateAdminRights'?

File: src/main/react-front-end/src/views/Admin/ClientsInfoComponent.js [hunk 2]
```diff
@@ -37,6 +44,41 @@ export default class ClientsInfoComponent extends Component{
...
+		else{
+			popupMsg = "Do you want to disable "+ row.firstName + " " + row.lastName+ " as admin.";
+			this.setState({showIsAdminPopup: true, adminChangePopupMsg:popupMsg, targetUser: row.email, isAdmin: false, firstName: row.firstName, lastName: row.lastName});
+		}		
+	}
+
+	// The actual call to update the admin information is triggered after the user selects "Yes" in the confirmation popup
+	// The user information is retrieved from the state
+	enableOrDisableAdminUsingStateInfo(email, isAdmin){
+		enableOrDisableAdmin(email, isAdmin).then((resp)=>{
```
Comment (LinusCastelino) [inline src/main/react-front-end/src/views/Admin/ClientsInfoComponent.js]: enableOrDisableAdmin() method name and signature similar to local method created on line 48.
Please rename for better code readability.

File: src/main/react-front-end/src/views/Admin/ClientsInfoComponent.js [hunk 3]
```diff
@@ -37,6 +44,41 @@ export default class ClientsInfoComponent extends Component{
 		});
 	}
 
+	// Shows the user a confirmation popup to confirm the update request.
+	enableOrDisableAdmin(event, row){
+		var popupMsg = ""
+		var isAdmin = event.target.checked;
+		if(isAdmin){
+			popupMsg = "Do you want to enable "+ row.firstName + " " + row.lastName+ " as admin.";
```
Comment (LinusCastelino) [inline src/main/react-front-end/src/views/Admin/ClientsInfoComponent.js]: Please update the message to
'Please confirm if ______ _______ must be granted admin privileges.'

File: src/main/react-front-end/src/views/Admin/ClientsInfoComponent.js [hunk 4]
```diff
@@ -37,6 +44,41 @@ export default class ClientsInfoComponent extends Component{
...
+			this.setState({showIsAdminPopup: true, adminChangePopupMsg:popupMsg, targetUser: row.email, isAdmin: true, firstName: row.firstName, lastName: row.lastName});
+		}
+		else{
+			popupMsg = "Do you want to disable "+ row.firstName + " " + row.lastName+ " as admin.";
```
Comment (LinusCastelino) [inline src/main/react-front-end/src/views/Admin/ClientsInfoComponent.js]: Please update this message to
''Please confirm if admin privileges of _______ ________ must be revoked.''

**Review by LinusCastelino — CHANGES_REQUESTED** [review]
Added review comments.
Please incorporate suggested changes.

File: src/main/java/org/onedatashare/server/controller/UserController.java (hunk 1 above)
Comment (KiranPrabhakar135) [reply src/main/java/org/onedatashare/server/controller/UserController.java]: Incorporated.

File: src/main/react-front-end/src/views/Admin/ClientsInfoComponent.js (hunk 2 above)
Comment (KiranPrabhakar135) [reply src/main/react-front-end/src/views/Admin/ClientsInfoComponent.js]: Incorporated. Renamed to enableOrDisableAdminApiCall

File: src/main/react-front-end/src/views/Admin/ClientsInfoComponent.js (hunk 3 above)
Comment (KiranPrabhakar135) [reply src/main/react-front-end/src/views/Admin/ClientsInfoComponent.js]: Incorporated

File: src/main/react-front-end/src/views/Admin/ClientsInfoComponent.js (hunk 4 above)
Comment (KiranPrabhakar135) [reply src/main/react-front-end/src/views/Admin/ClientsInfoComponent.js]: Incorporated

**Review by LinusCastelino — COMMENTED** [review]
Perfect!
Looks good now

**Pull request merged**